### ISSUE TITLE
(f . g)  <$>  x   to    f . g  <$>  x

### ISSUE THREAD

**Issue by Average-user** [ndmitchell/hlint#401]
HLint doesn't do this suggestion ```(f . g) <$> x ``` to ```f . g <$> x```

**Comment by Peaker** [comment]
Another reproduction:
```
x = undefined

infixl 1 ??
infixl 8 ^.

(??) = undefined
(^.) = undefined

a = x ?? (x ^. x)
```
(The brackets are redundant, but not reported).

**Comment by ndmitchell** [comment]
Generally HLint doesn't make assumptions about relative fixies for avoiding brackets - often more brackets than are strictly necessary giving the relative fixies is helpful for humans reading the code. As such the `x ?? (x ^. x)` example is one that I don't think HLint _should_ detect. However, the `foo . bar <$> baz` pattern is common enough that I think it probably should be spotted as a pattern rule. However, the current HLint patterns aren't quite powerful enough to detect that (they automatically see through brackets, making it hard to explicitly require a bracket on the LHS), so that needs tweaking first.

**Comment by ndmitchell** [comment]
Rather than tweak the pattern rules, I just added it as an explicitly coded rule - much easier.
